src/EditorFeatures/Core.Wpf/InlineRename/UI/Adornment/RenameFlyout.xaml.cs

```diff
@@ -221,56 +237,59 @@ private void Adornment_ConsumeMouseEvent(object sender, MouseButtonEventArgs e)
 
         private void Adornment_GotKeyboardFocus(object sender, KeyboardFocusChangedEventArgs e)
         {
-            if (e.OldFocus == this)
+            if (e.NewFocus != RenameUserInput)
             {
-                return;
+                RenameUserInput.Focus();
+                e.Handled = true;
             }
-
-            IdentifierTextBox.Focus();
-            e.Handled = true;
         }
 
         private void ToggleExpand(object sender, RoutedEventArgs e)
         {
             _viewModel.IsExpanded = !_viewModel.IsExpanded;
         }
 
+        private void RenameUserInput_GotFocus(object sender, RoutedEventArgs e)
+        {
+            this.RenameUserInput.SelectAllText();
+        }
+
         /// <summary>
         /// Respond to selection/cursor changes in the textbox the user is editing by
         /// applying the same selection to the textview that initiated the command
         /// </summary>
-        private void IdentifierTextBox_SelectionChanged(object sender, RoutedEventArgs e)
+        private void RenameUserInput_TextSelectionChanged(object sender, RoutedEventArgs e)
```

Comment:
Same here. This probably can be moved into the control itself since it already gets the `_viewModel` on creation

Reply:
Mostly, keeping the control behavior in the control makes it easier to follow IMO

Reply:
Also for `RenameUserInput_PreviewKeyDown` since it doesn't use anything unique to this control

Reply:
Same thing, I prefer keeping it here so we don't duplicate this logic into `SmartRenameUserInputComboBox` and `RenameUserInputTextBox`. I'm also not fan of putting it in `RenameUserInputPresenter` as I feel like we will spread logic across multiple area that might make the maintenance harder.